alicloud/table_alicloud_ram_policy.go

```diff
@@ -0,0 +1,245 @@
+package alicloud
+
+import (
+	"context"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/errors"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ram"
+	"github.com/sethvargo/go-retry"
+
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAlicloudRamPolicy(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:             "alicloud_ram_policy",
+		Description:      "Alibaba Cloud RAM Policy",
+		DefaultTransform: transform.FromCamel(),
+		List: &plugin.ListConfig{
+			Hydrate: listRAMPolicy,
+		},
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.AllColumns([]string{"name", "policy_type"}),
+			ShouldIgnoreError: isNotFoundError([]string{"InvalidParameter.PolicyType", "EntityNotExist.Policy", "MissingParameter"}),
+			Hydrate:           getRAMPolicy,
+		},
+		Columns: []*plugin.Column{
+			{
+				Name:        "name",
+				Type:        proto.ColumnType_STRING,
+				Description: "The name of the policy.",
+				Transform:   transform.FromField("PolicyName", "Policy.PolicyName"),
+			},
+
+			{
+				Name:        "policy_type",
+				Type:        proto.ColumnType_STRING,
+				Description: "The type of the policy. Valid values: System and Custom.",
+				Transform:   transform.FromField("PolicyType", "Policy.PolicyType"),
+			},
+			{
+				Name:        "description",
+				Type:        proto.ColumnType_STRING,
+				Description: "The policy description",
+				Transform:   transform.FromField("Description", "Policy.Description"),
+			},
+			{
+				Name:        "default_version",
+				Type:        proto.ColumnType_STRING,
+				Description: "Deafult version of the policy",
+				Transform:   transform.FromField("DefaultVersion", "Policy.DefaultVersion"),
+			},
+			{
+				Name:        "create_date",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Description: "Policy creation date",
+				Transform:   transform.FromField("CreateDate", "Policy.CreateDate"),
+			},
+			{
+				Name:        "update_date",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Description: "Last time when policy got updated ",
+				Transform:   transform.FromField("UpdateDate", "Policy.UpdateDate"),
+			},
+			{
+				Name:        "attachment_count",
+				Type:        proto.ColumnType_INT,
+				Description: "The number of references to the policy.",
+				Transform:   transform.FromField("AttachmentCount", "Policy.AttachmentCount"),
+			},
+			{
+				Name:        "version_id",
```

Comment:
When listing RAM policies, would we ever get non-default policy versions? If so, does Alicloud API always return them under `DefaultPolicyVersion.VersionId`?

Reply:
we will always get the defaultPolicy details.

Reply:
yes, will update